### PR TITLE
Add Query methods to builder for use with standard database/sql package

### PR DESCRIPTION
**Pull request by icholy**
The query methods generate a query & args that can be used by the `database/sql` package.

``` go
query, args := lk.Select("*").From("foo").Query()
rows, err := db.Query(query, args...)
```

This addresses #26 but it's also useful with sqlx (which I'm using). It lets you use the `sqlx.DB.Select` and `sqlx.DB.Get` methods without creating a statement (which is clunky).

### CODE REVIEW

**Comment by novln** [comment]
Hi,

Thank you for this pull request, I really like the idea.
However, I would like some modifications before merging it.

https://github.com/icholy/loukoum/blob/f51430f70e81bda8d45b29ffaf8ac08d7eb3f0e8/types/context.go#L31

Appending twice here is not optimal. We could convert `Context` into an interface.

```go
type Context struct {
	Write(query string)
	Bind(value interface{})
	Query() string
	Values() map[string]interface{} 
	Args() []interface{}
}
```

And by defining two types:

```go
type ContextType string
const (
    StdContextType = ContextType("standard")
    NamedContextType = ContextType("named")
)
```

That could be given as an optional argument for `NewContext`:

```go
func NewContext(kind ...ContextType) Context
```

_Default being `NamedContextType`..._

We could have two struct that has different behavior.

Thank you :slightly_smiling_face:

**Comment by icholy** [comment]
@novln I added an extra context that the String() method can use, and the other embed. Also, I got rid of the constructors.

**Comment by icholy** [comment]
I'm not sure what this error means in the ci
```
types/context.go:1::warning: file is not goimported (goimports)
```

**Comment by icholy** [comment]
I think it also makes sense to rename `Builder.Prepare` to `Builder.NamedQuery`. We should leave `Builder.Prepare` where it is for now, but mark is as deprecated and proxy it to `Builder.NamedQuery`.

**Comment by novln** [comment]
Hi,

I really like the `RawContext`, `NamedContext` and `StdContext` idea. 
However can you remove the `String()` method from the interface, it would expose a much simpler API.

Also, since this pull request breaks API compatibility can you change the branch to `dev`, I will ship this in a v2 of loukoum. 
And since it's a new version, you can rename `Builder.Prepare` to `Builder.NamedQuery` without having to do the proxy.

Thank you.

PS: I'll deal with the linter later.

File: types/context.go [hunk 1]
```diff
@@ -3,44 +3,75 @@ package types
 import (
 	"bytes"
 	"fmt"
+	"github.com/ulule/loukoum/format"
```
Comment (novln) [inline types/context.go]: Could you add a newline here ?
Sorry to be nitpicking, it's just our internal code style at Ulule.

```go
import (
	"bytes"
	"fmt"

	"github.com/ulule/loukoum/format"
)
```

**Comment by icholy** [comment]
Are you referring to the String method on the builders?

**Comment by novln** [comment]
No sorry, the `String()` method on the `Context` interface.

**Comment by icholy** [comment]
I don't think there is one

**Comment by novln** [comment]
You right, I meant `Query()`

**Comment by icholy** [comment]
Ah ok, I see what you mean.

**Comment by icholy** [comment]
@novln I think it's good now.

**Comment by novln** [comment]
Thank you.

**Pull request merged**